packages/@snowpack/plugin-webpack/plugin.js

```diff
@@ -10,8 +11,130 @@ const CopyPlugin = require("copy-webpack-plugin");
+function insertBefore(newNode, existingNode) {
+  existingNode.parentNode.insertBefore(newNode, existingNode);
+}
+
+function parseHTMLFiles({ srcDirectory }) {
+  // Get all html files from the output folder
+  const pattern = srcDirectory + "/**/*.html";
+  const htmlFiles = glob
+    .sync(pattern)
+    .map((htmlPath) => path.relative(srcDirectory, htmlPath));
+
+  const doms = {};
+  const jsEntries = {};
+  for (const htmlFile of htmlFiles) {
+    const dom = new JSDOM(fs.readFileSync(path.join(srcDirectory, htmlFile)));
+
+    //Find all local script, use it as the entrypoint
+    const scripts = Array.from(dom.window.document.querySelectorAll("script"))
+      .filter((el) => el.type.trim().toLowerCase() === "module")
+      .filter((el) => !/^[a-zA-Z]+:\/\//.test(el.src));
+
+    for (const el of scripts) {
+      const src = el.src.trim();
+      const parsedPath = path.parse(src);
+      const name = parsedPath.name;
+      if (!(name in jsEntries)) {
+        jsEntries[name] = {
+          path: path.join(srcDirectory, src),
+          occurrences: [],
+        };
+      }
+      jsEntries[name].occurrences.push({ script: el, dom });
+    }
+
+    doms[htmlFile] = dom;
+  }
+  return { doms, jsEntries };
+}
+
+function emitHTMLFiles({ doms, jsEntries, stats, baseUrl, destDirectory }) {
+  const entrypoints = stats.toJson({ assets: false, hash: true }).entrypoints;
+
+  //Now that webpack is done, modify the html files to point to the newly compiled resources
+  Object.keys(jsEntries).forEach((name) => {
+    if (entrypoints[name] !== undefined && entrypoints[name]) {
+      const assetFiles = entrypoints[name].assets || [];
+      const jsFiles = assetFiles.filter((d) => d.endsWith(".js"));
+      const cssFiles = assetFiles.filter((d) => d.endsWith(".css"));
+
+      for (const occurrence of jsEntries[name].occurrences) {
+        const originalScriptEl = occurrence.script;
+        const dom = occurrence.dom;
+        const head = dom.window.document.querySelector("head");
+
+        for (const jsFile of jsFiles) {
+          const scriptEl = dom.window.document.createElement("script");
+          scriptEl.src = path.posix.join(baseUrl, jsFile);
+          // insert _before_ so the relative order of these scripts is maintained
+          insertBefore(scriptEl, originalScriptEl);
+        }
+        for (const cssFile of cssFiles) {
+          const linkEl = dom.window.document.createElement("link");
+          linkEl.setAttribute("rel", "stylesheet");
+          linkEl.href = path.posix.join(baseUrl, cssFile);
+          head.append(linkEl);
+        }
+        originalScriptEl.remove();
+      }
+    }
+  });
+
+  //And write our modified html files out to the destination
+  for (const [htmlFile, dom] of Object.entries(doms)) {
+    fs.writeFileSync(path.join(destDirectory, htmlFile), dom.serialize());
+  }
+}
+
+function getSplitChunksConfig({ numEntries }) {
+  return {
```

Comment:
minor: would you mind adding a comment here linking to https://web.dev/granular-chunking-nextjs/ (or wherever is a good reference for this)? It’d help future contributors understand some of the settings.